vllm/distributed/device_communicators/custom_all_reduce.py

```diff
@@ -264,11 +264,20 @@ def capture(self):
                 self.register_graph_buffers()
 
     def _get_ipc_meta(self, inp: torch.Tensor):
-        data = inp.untyped_storage()._share_cuda_()
-        shard_data = (
-            data[1],  # ipc handle to base ptr
-            data[3],  # offset of base ptr
-        )
+        if is_hip():
+            # _share_cuda_() doesn't accept meta buffer not allocated from
+            # PyTorch cache allocator, use direct HIP call to get IPC handle
```

Comment:
@gshtras , @mawong-amd  make sure we have this fix, not docker related, in our main.